analytics/service.go

```diff
@@ -0,0 +1,111 @@
+package analytics
+
+import (
+	"fmt"
+	"github.com/ONSdigital/go-ns/log"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const pageIndexParam = "pageIndex"
+const pageSizeParam = "pageSize"
+const linkIndexParam = "linkIndex"
+const urlParam = "url"
+const termParam = "term"
+const searchTypeParam = "type"
+const timestampKey = "timestamp"
+
+// Service - defines a Stats Service Interface
+type Service interface {
+	CaptureAnalyticsData(analytics *Model) string
+}
+
+// ServiceImpl - Implementation of the Analytics Service interface.
+type ServiceImpl struct{}
+
+// Model - Type to encapsulate Search Analytics data.
+type Model struct {
+	url        string
+	term       string
+	searchType string
+	pageSize   int
+	pageIndex  int
+	linkIndex  int
+}
+
+// NewServiceImpl - Creates a new Analytics ServiceImpl.
+func NewServiceImpl() *ServiceImpl {
+	return &ServiceImpl{}
+}
+
+// GetURL - Get the URL of the search result the user clicked.
+func (a *Model) GetURL() string {
```

Comment:
don't think any of these accessor functions are needed - can't see them being used anywhere (except in tests), we always refer to the fields directly as `.url` etc